src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,124 @@
+package duke;
+
+import duke.command.AddCommand;
+import duke.command.ClearCommand;
+import duke.command.Command;
+import duke.command.DoneCommand;
+import duke.command.ExitCommand;
+import duke.command.FindCommand;
+import duke.command.ListCommand;
+import duke.command.RemoveCommand;
+import duke.command.RestoreCommand;
+import duke.task.TaskTypes;
+
+/**
+ * The Parser class encapsulates methods related to parsing user input.
+ */
+public class Parser {
+
+    /**
+     * Parses a given string representing user input and returns the related Command.
+     * Throws a DukeException when an invalid command is received.
+     *
+     * @param str The given user input.
+     * @return a Command corresponding to the user input.
+     * @throws DukeException when the given user input is invalid.
+     */
+    public static Command parse(String str) throws DukeException {
+        String[] strArr = str.split("\\s", 2);
+
+        switch (strArr[0]) {
```

Comment:
I really like the use of switch statements to handle exceptional inputs and re-direct the input to the necessary command methods, really helps make the code look neat and understandable.